test/stringio/test_stringio.rb

```diff
@@ -237,8 +237,9 @@ def test_write_encoding_conversion
 
   def test_write_integer_overflow
     f = StringIO.new
-    f.pos = RbConfig::LIMITS["LONG_MAX"]
     assert_raise(ArgumentError) {
+      # JRuby errors when setting pos to an out-of-range value
+      f.pos = RbConfig::LIMITS["LONG_MAX"]
```

Comment:
Is there any plan to fix it?

Reply:
This seems to be because `ByteList` cannot handle `long`s.
That means JRuby cannot make 4GiB `StringIO`?

Reply:
That is correct; there's no plan to fix this because the JVM itself cannot manage arrays larger than a signed integer.

The test change here is due to JRuby detecting `pos=` set to a value above this size and immediately raising. I think CRuby should also add this behavior for `pos=` above the maximum range of an array offset.